test/render/TextRenderer.test.js

```diff
@@ -0,0 +1,60 @@
+define([
+    'src/util/Font',
+    'src/render/TextRenderer',
+    'src/geom/Vec2'
+], function (Font,
+             TextRenderer,
+             Vec2) {
+    "use strict";
+    describe("TextRenderer tests", function () {
+
+        // Mocking TextRenderer.textSize() to avoid 2D context requirement.
+        TextRenderer.prototype.textSize = function (text, font, outline) {
+            return new Vec2(text.length * 7, 16);
+        };
+
+        var testText = "Lorem ipsum dolor sit amet, consectetur "
+            + "adipiscing elit, sed do eiusmod tempor incididunt ut";
+
+        var myFont = new Font(15);
+
+        it("Should throw an exception on missing text input", function () {
+            expect(function () {
+                var mockTextRenderer = new TextRenderer();
+                mockTextRenderer.wrap(null, 20, 100, myFont);
+            }).toThrow();
+        });
+
+        it("Should output '...' due to wrap height being less than textSize height", function () {
+            var mockTextRenderer = new TextRenderer();
+            var wrappedText = mockTextRenderer.wrap(testText, testText.length, 15, myFont);
```

Comment:
While not really a problem, using testText.length for width here and on line 36 is misleading because the width parameter is the number of pixels to allow. I would change this to a constant.